core/src/main/kotlin/Formats/MarkdownFormatService.kt

```diff
@@ -44,44 +118,57 @@ open class MarkdownFormatService(locationService: LocationService,
     }
 
     override fun formatLink(text: String, href: String): String {
-        return "[$text]($href)"
+        return """[${text.replace("[", "\\[").replace("]","\\]")}](${href})"""
     }
 
     override fun appendLine(to: StringBuilder, text: String) {
-        to.appendln(text)
+        to.appendWithAdjustedLF(text)
+    }
+
+    override fun appendList(to: StringBuilder, body: () -> Unit) {
+        to.append(allowedLf().requiredWhitespace)
+        changeLf(LfPriority.HtmlBr) { body() }
     }
 
     override fun appendAnchor(to: StringBuilder, anchor: String) {
         // no anchors in Markdown
     }
 
     override fun appendParagraph(to: StringBuilder, text: String) {
-        to.appendln()
-        to.appendln(text)
-        to.appendln()
+        to.append(allowedLf().paragraph.start)
+        to.append(text)
+        to.append(allowedLf().paragraph.end)
     }
 
     override fun appendHeader(to: StringBuilder, text: String, level: Int) {
-        appendLine(to)
-        appendLine(to, "${"#".repeat(level)} $text")
-        appendLine(to)
-    }
-
-    override fun appendBlockCode(to: StringBuilder, lines: List<String>, language: String) {
-        appendLine(to)
-        to.appendln(if (language.isEmpty()) "```" else "``` $language")
-        to.appendln(lines.joinToString("\n"))
-        to.appendln("```")
-        appendLine(to)
-    }
-
-    override fun appendTable(to: StringBuilder, body: () -> Unit) {
-        to.appendln()
-        body()
-        to.appendln()
-    }
-
-    override fun appendTableHeader(to: StringBuilder, body: () -> Unit) {
+        to.append(allowedLf().requiredWhitespace)
+        to.appendWithAdjustedLF("${"#".repeat(level)} $text")
+    }
+
+    override fun appendBlockCode(to: StringBuilder, language: String, bodyAsLines: ()->List<String>) {
+        to.append(allowedLf().requiredWhitespace)
+
+        val codeTags = allowedLf().inlineCode
+        if (codeTags.start.startsWith('`')) {
+            changeLf(LfPriority.Code) {
+                // this does not adjust line feeds, a code block in a list or table will just break formatting
+                to.appendln(if (language.isEmpty()) "```" else "``` $language")
+                // use 4 zero width spaces which act as a code fence escaping any use of ``` within the code block
+                to.appendln(bodyAsLines().map { "​\u200B\u200B\u200B\u200B$it" }.joinToString("\n"))
```

Comment:
Ah code fencing.  It is because if you use backticks inside of code it can break the code block.  

You can create a code block alternatively using 4 white-space characters.  

So by using 4 zero space white space characters it ensures no matter what fencing is used, that backticks within the code block are ignored.

It is basically escaping everything each code block line.  

It is a work around until the "add more back ticks per fence" method is supported by more markdown renderers.